src/main/java/com/epam/petclinic/clinic/controller/ClinicController.java

```diff
@@ -0,0 +1,71 @@
+package com.epam.petclinic.clinic.controller;
+
+import com.epam.petclinic.clinic.model.Clinic;
+import com.epam.petclinic.clinic.model.Offer;
+import com.epam.petclinic.clinic.repository.ClinicRepository;
+import com.epam.petclinic.clinic.repository.OfferRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+/**
+ * Clinic controller.
+ * <p>
+ * Date: 2017-04-13
+ *
+ * @author Stanislau_Halauniou
+ */
+@RestController
+@RequestMapping("/api/clinics")
+class ClinicController {
+
+    @Autowired
+    private ClinicRepository clinicRepository;
+
+    @Autowired
+    private OfferRepository offerRepository;
+
+    //TODO: add transaction, investigate cascading
+    @RequestMapping(value = "/", method = RequestMethod.POST)
+    public Clinic create(@RequestBody Clinic clinic) {
+        Clinic createdClinic = clinicRepository.save(clinic);
+        List<Offer> offers = clinic.getOffers();
+        offers.stream().forEach(offer -> offer.setClinic(clinic));
+        offerRepository.save(offers);
+        clinic.setId(createdClinic.getId());
+        return clinic;
+    }
+
+    //TODO: add transaction, investigate cascading
+    @RequestMapping(value = "/{id}", method = RequestMethod.PUT)
+    public Clinic update(@RequestBody Clinic clinic, @PathVariable String id) {
```

Comment:
Let's update only clinic fields (address, name and so on) and manage association between clinic and offer by 2 different calls (add/remove clinic offer)